pkg/tempopb/backendwork.proto

```diff
@@ -46,8 +48,7 @@ message JobDetail {
   string tenant = 1;
```

Comment:
Currently the retention job retains all tenants, and so doesn't need the tenant field.  Perhaps `tenant` could move to the job detail on the jobs which need it.